src/ComponentRect.js

```diff
@@ -1,45 +1,25 @@
 /* eslint-disable require-jsdoc */
 import React from "react";
-import { Rect } from "react-konva";
-import { MatrixCell, ConnectorRect } from "./ComponentConnectorRect";
+import {Rect} from "react-konva";
+import {ConnectorRect, MatrixCell} from "./ComponentConnectorRect";
 import PropTypes from "prop-types";
-
-const zip = (arr, ...arrs) => {
-  /*Credit: https://gist.github.com/renaudtertrais/25fc5a2e64fe5d0e86894094c6989e10*/
-  return arr.map((val, i) => arrs.reduce((a, arr) => [...a, arr[i]], [val]));
-};
+import {sum} from "./utilities";
 
 export function compress_visible_rows(components) {
   /*Returns a Map with key of the original row number and value of the new, compressed row number.
    * Use this for y values of occupancy and LinkColumn cells.  */
-  let rows_present = find_rows_visible_in_viewport(components);
+  let all_visible = new Set();
+  for (let c of components) {
+    c.occupants.map((row) => all_visible.add(row));
```

Comment:
Is WebStorm not complaining that the vector returned by map() is not considered?

Reply:
+1, looks like a `.forEach` candidate

Reply:
I've just pushed a new commit with this fix.  Ready to merge?  @AndreaGuarracino  it's your call.